lib/efi_loader/efi_capsule.c

```diff
@@ -632,6 +632,8 @@ static efi_status_t find_boot_device(void)
 		ret = get_dp_device(boot_var16, &boot_dev);
 		if (ret == EFI_SUCCESS) {
 			if (device_is_present_and_system_part(boot_dev)) {
+			/* If device is present retrieveing the file node before exiting*/
+				bootdev_root = efi_file_from_path(boot_dev);
 				goto out;
```

Comment:
This is touching common code, so you need to defend why the change is necessary. You also need to be sure that changing this won't break other platforms.

Why is this change necessary?

Reply:
/* If device is present retrieving the file node before exiting*/
To handle the device present scenario we are checking if device present and updating the device handle.
If efi partition device already opened boot_root will be null and the device will lead to crash.